src/main/sf_496C0.c

```diff
@@ -1,71 +1,2995 @@
+            Math_SmoothStepToF(&obj2F4->unk_0F4.z, 0.0f, 0.05f, 0.2f, 0.0001f);
+            break;
+        case 1:
+            obj2F4->unk_0B8 = 2;
+            Audio_PlaySfx(0x09000002U, &obj2F4->sfxPos, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            obj2F4->timer_0BC = 0x96;
+            obj2F4->unk_188 = 5.0f;
+        case 2:
+            obj2F4->unk_07C = 2;
+            obj2F4->unk_114 += 2.0f;
+            if (obj2F4->timer_0BC == 0) {
+                Object_Kill(&obj2F4->obj, &obj2F4->sfxPos);
+            }
+            break;
+    }
+    Matrix_RotateY(gCalcMatrix, (obj2F4->unk_0F4.y + 180.0f) * M_DTOR, 0);
+    Matrix_RotateX(gCalcMatrix, -(obj2F4->unk_0F4.x * M_DTOR), 1);
+    sp3C.x = 0.0f;
+    sp3C.y = 0.0f;
+    sp3C.z = obj2F4->unk_114;
+    Matrix_MultVec3fNoTranslate(gCalcMatrix, &sp3C, &sp30);
+    obj2F4->unk_0E8.x = sp30.x;
+    obj2F4->unk_0E8.y = sp30.y;
+    obj2F4->unk_0E8.z = sp30.z;
+    obj2F4->obj.rot.x = -obj2F4->unk_0F4.x;
+    obj2F4->obj.rot.y = obj2F4->unk_0F4.y + 180.0f;
+    obj2F4->obj.rot.z = -obj2F4->unk_0F4.z;
+}
+
+void func_8004990C(Player* player) {
+    if (D_801784AC == 0) {
+        D_80177CE8 += 60.0f;
+    }
+    player->unk_138 = player->pos.z + player->unk_08C;
+    player->unk_0F8 = player->unk_0EC + player->unk_12C + player->unk_130;
+}
+
+void func_80049968(Object_2F4* arg0, s32 arg1) {
+    Object_2F4_Initialize(arg0);
+    arg0->obj.status = 1;
+    arg0->obj.id = 0xC3;
+    arg0->obj.pos.x = D_800C9FD0[arg1];
+    arg0->obj.pos.y = D_800C9FE0[arg1];
+    arg0->obj.pos.z = D_800C9FF0[arg1] + (((void) 0, gPlayer))->unk_138;
+    arg0->unk_114 = D_800CA000[arg1];
+    arg0->unk_118 = D_800CA010[arg1];
+    arg0->unk_0B6 = D_800CA030[arg1];
+    arg0->unk_0B8 = D_800CA040[arg1];
+    arg0->obj.rot.z = D_800CA020[arg1];
+    arg0->obj.rot.y = 180.0f;
+    arg0->unk_0E8.z = gPlayer->vel.z;
+    Object_SetInfo(&arg0->info, arg0->obj.id);
+    arg0->unk_07C = 1;
+    Audio_PlaySfx(0x3100000CU, &arg0->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+}
+
+void func_80049A9C(Object_8C* obj8c, f32 x, f32 y, f32 z) {
+    Object_8C_Initialize(obj8c);
+    obj8c->obj.status = 1;
+    obj8c->obj.id = 0x15A;
+    obj8c->timer_50 = 0x64;
+    obj8c->scale2 = 0.2f;
+    obj8c->obj.pos.x = x;
+    obj8c->obj.pos.y = y;
+    obj8c->obj.pos.z = z;
+    obj8c->unk_54.z = 40.0f;
+    obj8c->obj.rot.z = Rand_ZeroOne() * 360.0f;
+    Object_SetInfo(&obj8c->info, obj8c->obj.id);
+}
+
+void func_80049B44(void) {
+    s32 i;
+    f32 x;
+    f32 y;
+    f32 z;
+
+    for (i = 0; i < ARRAY_COUNT(gObjects8C); i++) {
+        if (!gObjects8C[i].obj.status) {
+            x = (Rand_ZeroOne() - 0.5f) * 400.0f;
+            y = (Rand_ZeroOne() - 0.5f) * 400.0f;
+            z = (-D_80177D20 - 500.0f) - Rand_ZeroOne() * 500.0f;
+            func_80049A9C(&gObjects8C[i], x, y, z);
+            break;
+        }
+    }
+}
+
+void func_80049C0C(Player* arg0) {
+    s32 var_v0;
+    char pad[0x10];
```

Comment:
pads should be s32 when possible.
```suggestion
s32 pad[4];